### PR TITLE
tstest/integration: add retry loops around tailscaled startup

### PR DESCRIPTION
**Pull request by josharian**
On my mac machine,

go test -race -count=1 ./tstest/integration

failed consistently. The failures were all around failure to communicate
with tailscaled. tailscale could connect to tailscaled, but got EOFs
in response to requests.

These failures are transient, and are probably related to slow
tailscaled startup. (Note that we propagate the -race flag
to tailscale and tailscaled.)

Adding a retry loop around early tailscaled communication
stabilizes the test.

Signed-off-by: Josh Bleecher Snyder <josh@tailscale.com>

### CODE REVIEW

File: tstest/integration/integration_test.go [hunk 1]
```diff
@@ -701,8 +701,12 @@ func (n *testNode) MustUp(extraArgs ...string) {
 	}
 	args = append(args, extraArgs...)
 	t.Logf("Running %v ...", args)
-	if b, err := n.Tailscale(args...).CombinedOutput(); err != nil {
```
Comment (bradfitz) [inline tstest/integration/integration_test.go:704]: the tailscale CLI command already has a loop waiting for the server to come up. Does that not work on macOS?

Comment (josharian) [reply tstest/integration/integration_test.go:704]: The loop is waiting to make a connection. In this case, we've successfully connected, but got an EOF in response to the request we made:

```
=== CONT  TestOneNodeUp_NoAuth
    integration.go:75: built [tailscale.com/cmd/tailscaled tailscale.com/cmd/tailscale] in 1.196s
    integration_test.go:463: DERP httpsrv listener: 127.0.0.1:57707
2021/12/02 13:31:01 control: mapRoutine: backoff: 90 msec
2021/12/02 13:31:02 You have enabled a non-default log target. Doing without being told to by Tailscale staff or your network administrator will make getting support difficult.
2021/12/02 13:31:02 logtail started
2021/12/02 13:31:02 Program starting: v1.19.0-date.20211101, Go 1.17.3-ts7b34cacdfc-race: []string{"/var/folders/x7/n21v1sbs035fdbjjyq347xfc0000gn/T/TestOneNodeUp_NoAuth3229572814/001/tailscaled", "--tun=userspace-networking", "--state=/var/folders/x7/n21v1sbs035fdbjjyq347xfc0000gn/T/TestOneNodeUp_NoAuth3229572814/002/tailscale.state", "--socket=/var/folders/x7/n21v1sbs035fdbjjyq347xfc0000gn/T/TestOneNodeUp_NoAuth3229572814/002/tailscale.sock", "--socks5-server=localhost:0"}
2021/12/02 13:31:02 LogID: d7e08a6e6a78ec155a3c982ee5a92b71c182724f2264f35ec980e5a4746d30f0
2021/12/02 13:31:02 logpolicy.Read /var/folders/x7/n21v1sbs035fdbjjyq347xfc0000gn/T/TestOneNodeUp_NoAuth3229572814/003/tailscaled.log.conf: open /var/folders/x7/n21v1sbs035fdbjjyq347xfc0000gn/T/TestOneNodeUp_NoAuth3229572814/003/tailscaled.log.conf: no such file or directory
2021/12/02 13:31:02 magicsock: home is now derp-1 (test)
2021/12/02 13:31:02 logtail: dialed "127.0.0.1:57710" in 1ms
2021/12/02 13:31:02 SOCKS5 listening on 127.0.0.1:57714
2021/12/02 13:31:02 magicsock: endpoints changed: 127.0.0.1:59486 (stun), 192.168.4.37:59486 (local)
2021/12/02 13:31:02 magicsock: adding connection to derp-1 for home-keep-alive
2021/12/02 13:31:02 magicsock: 1 active derp conns: derp-1=cr0s,wr0s
2021/12/02 13:31:02 wgengine.NewUserspaceEngine(tun "userspace-networking") ...
2021/12/02 13:31:02 dns: using dns.noopManager
2021/12/02 13:31:02 link state: interfaces.State{defaultRoute=en7 ifs={en7:[192.168.4.37/22] utun5:[100.124.5.51/32 fd7a:115c:a1e0:ab12:4843:cd96:627c:533/48]} httpproxy=http://127.0.0.1:57711 v4=true v6=false}
2021/12/02 13:31:02 magicsock: disco key = d:9dcdfc2fff9d5179
2021/12/02 13:31:02 Creating wireguard device...
2021/12/02 13:31:02 Bringing wireguard device up...
2021/12/02 13:31:02 Bringing router up...
2021/12/02 13:31:02 Clearing router settings...
2021/12/02 13:31:02 Starting link monitor...
2021/12/02 13:31:02 Engine created.
2021/12/02 13:31:02 control: NetInfo: NetInfo{varies= hairpin=true ipv6=false udp=true derp=#1 portmap=U link=""}
2021/12/02 13:31:02 derphttp.Client.Connect: connecting to derp-1 (test)
2021/12/02 13:31:02 control: client.newEndpoints(0, [127.0.0.1:59486 192.168.4.37:59486])
2021/12/02 13:31:02 Switching ipn state Starting -> Running (WantRunning=true, nm=true)

=== CONT  TestOneNodeUp_NoAuth
    integration_test.go:73: running tailscale status: exit status 1, failed to connect to local tailscaled (which appears to be running). Got error: Get "http://local-tailscaled.sock/localapi/v0/status": EOF
        
    stuntest.go:64: STUN server shutdown
--- FAIL: TestOneNodeUp_NoAuth (6.27s)
```

Comment (josharian) [reply tstest/integration/integration_test.go:704]: OK, another data point. The tests only fail if I'm also running Tailscale.app (not tailscaled).

Comment (josharian) [reply tstest/integration/integration_test.go:704]: Aha! cmd/tailscale is smart enough to find Tailscale.app. So what happens is that testNode.AwaitRunning asks for a status and gets one, even though the tailscaled we just launched isn't actually running yet. 🤦🏽 

Maybe we can turn off "talk to Tailscale.app" when the `--socket` flag is provided. Looking...

**Comment by josharian** [comment]
Better fix: #3499

**Pull request closed without merging**